File: create_library_with_header/use_library/src/use_library.cpp
#include "rclcpp/rclcpp.hpp"
#include "publisher_library/timed_publisher.hpp"

class UseLibrary : public rclcpp::Node
{
public:
    UseLibrary() : Node("use_library")
    {
        RCLCPP_INFO(this->get_logger(), "UseLibrary node started !");

        publisher1_ = std::make_shared<publisher_library::TimedPublisher>(
            shared_from_this(),
            "topic_fast",
            0.5
        );
        publisher1_->setMessage("Fast message");

        publisher2_ = std::make_shared<publisher_library::TimedPublisher>(
            shared_from_this(),
            "topic_slow",
            2.0
        );
        publisher2_->setMessage("Slow message");
    }

private:
    std::shared_ptr<publisher_library::TimedPublisher> publisher1_;
    std::shared_ptr<publisher_library::TimedPublisher> publisher2_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<UseLibrary>());
    rclcpp::shutdown();
    return 0;
}

File: ros2_fundamentals/create_library_with_header_py/use_library_py/use_library_py/use_library.py
import rclpy
from rclpy.node import Node
from publisher_library_py.timed_publisher import TimedPublisher
#    | importer la librairie

class UseLibrary(Node):
    def __init__(self):
        super().__init__("use_library")

        self.get_logger().info("UseLibrary node started !")

        # --------------------------------------------------------
        # UTILISER LA LIBRAIRIE
        # --------------------------------------------------------
        # Publisher rapide
        self.publisher1_ = TimedPublisher(
            self,          #  le nœud actuel
            "topic_fast",  #  topic
            0.5            #  période 500ms
        )
        self.publisher1_.set_message("Fast message")

        # Publisher lent
        self.publisher2_ = TimedPublisher(
            self,
            "topic_slow",
            2.0   #  période 2s
        )
        self.publisher2_.set_message("Slow message")

def main(args=None):
    rclpy.init(args=args)
    node = UseLibrary()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()